### PR TITLE
feat: added option to disable key collisions

### PR DESCRIPTION
**Pull request by simlu**

### CODE REVIEW

**Comment by simlu** [comment]
Will be done in https://github.com/blackflux/object-scan/issues/303

File: README.md [hunk 1]
```diff
@@ -102,10 +106,17 @@ When set to false, joined paths for functions and the final result are not escap
 Type: `boolean`<br>
 Default: `true`
 
-When set to false, no array selectors should be used in any needles and arrays are automatically traversed.
+When set to `false`, no array selectors should be used in any needles and arrays are automatically traversed.
 
 Note that the results still include the array selectors.
 
+#### keyCollision
+
```
Comment (simlu) [inline README.md]: Rename to needleCollision?

File: test/util/compiler.spec.js [hunk 2]
```diff
@@ -77,6 +77,17 @@ describe('Testing compiler', () => {
     expect(compiler.isRecursive(tower.a)).to.equal(false);
   });
 
+  it('Testing recursion position on overwrite', () => {
```
Comment (simlu) [inline test/util/compiler.spec.js]: Needle overwrite?

File: README.md (hunk 1 above)
Comment (simlu) [reply README.md]: Or negate and call `neexleOverwrite`?

File: README.md [hunk 3]
```diff
@@ -102,10 +106,17 @@ When set to false, joined paths for functions and the final result are not escap
 Type: `boolean`<br>
 Default: `true`
 
-When set to false, no array selectors should be used in any needles and arrays are automatically traversed.
+When set to `false`, no array selectors should be used in any needles and arrays are automatically traversed.
 
 Note that the results still include the array selectors.
 
+#### keyCollision
```
Comment (MacMcIrish) [inline README.md]: `ignoreKeyCollision` (and negate of course)
`ignoreNeedleCollision`

Comment (MacMcIrish) [reply README.md]: `silenceNeedleCollision: true`

File: README.md (hunk 1 above)
Comment (simlu) [reply README.md]: `validate`
`validateNeedles`
`validateNeedlesUnique`
`enforceNeedlesUnique`
`verifyNeedlesUnique`

Comment (simlu) [reply README.md]: `hardCheck`
`strict`

**Pull request merged**